source/server/http/admin.h

```diff
@@ -164,9 +164,16 @@ class AdminImpl : public Admin,
   void writeClustersAsJson(Buffer::Instance& response);
   void writeClustersAsText(Buffer::Instance& response);
 
+  static bool shouldShowMetric(const std::shared_ptr<Stats::Metric> metric, const bool used_only,
+                               const absl::optional<std::regex> regex) {
+    return ((!used_only || metric->used()) &&
+            (!regex.has_value() || std::regex_match(metric->name(), regex.value())));
```

Comment:
question: would this be more user-friendly if we use std::regex_search instead? Then a filter of "server" would get all the server stats. And if you want to do a full-string match, add begin-line and end-line anchors.

Reply:
I like this idea a lot -- see 599e857.